### ISSUE TITLE
No polydispersity distributions for `core_multi_shell` model

### ISSUE THREAD

**Issue by lozanodorian** [SasView/sasview#2563]
**Describe the bug**
The Gaussian distribution is the only distribution available for `Distribution of thickness` in the `core_multi_shell` model. 

**To Reproduce**
1. In the fit panel, choose `sphere`, `core_multi_shell`,
2. Check `Option > Polydispersity`, then click on the `Polydispersity` tab,
3. See error: no drop-down menus for thicknesses.

**Expected behavior**
A drop-down menu in `Function` for each `thickness`.

**Screenshots**
![Screenshot from 2023-07-24 13-02-04](https://github.com/SasView/sasview/assets/125293676/f956a03f-1135-43c8-8017-f09517160187)

**Note**
For the remaining drop-down menu (thickness2, not radius), a choice of distribution can lead to the following error:
```
12:58:47 - ERROR: Traceback (most recent call last):
  File "sasview/src/sas/qtgui/Perspectives/Fitting/FittingWidget.py", line 3240, in <lambda>
    func.currentIndexChanged.connect(lambda: self.onPolyComboIndexChange(str(func.currentText()), i))
  File "sasview/src/sas/qtgui/Perspectives/Fitting/FittingWidget.py", line 3316, in onPolyComboIndexChange
    self.kernel_module.set_dispersion(param.name, self.disp_model)
  File "sasmodels/sasmodels/sasview_model.py", line 816, in set_dispersion
    raise ValueError("%r is not a dispersity or orientation parameter"
ValueError: 'thickness[n]' is not a dispersity or orientation parameter
```

**SasView version (please complete the following information):**
 - Version: 5.0.6

**Operating system (please complete the following information):**
 - OS: Linux

**Comment by smk78** [comment]
I think this behaviour is part of a bigger issue with what we term 'multiplicity models'. So, for example, if you use the core_multi_shell with n=1 the resulting model does not behave like the core_shell model (which _does_ allow a choice of dispersity function for the single shell).

**Comment by smk78** [comment]
Also see https://github.com/SasView/sasview/issues/2243

**Comment by butlerpd** [comment]
This issue should have been closed with the merge of #2647. Closing it manually
